builds/fabric8-che/assembly/user-auth-valve/src/main/java/com/redhat/che/valve/UserAuthValve.java

```diff
@@ -41,15 +46,13 @@
  * @author amisevsk
  */
 public class UserAuthValve extends KeycloakAuthenticatorValve {
-
+    
     private static final Log LOG = LogFactory.getLog(UserAuthValve.class);
-    private static final String USER_VALIDATOR_ENDPOINT = "http://che-host:8080/api/token/user";
+    private static final String API_ENDPOINT = "http://che-host:8080/api";
+    private static final String USER_VALIDATOR_ENDPOINT = API_ENDPOINT + "/token/user";
     private static final String AUTHORIZATION_HEADER = "Authorization";
-    private static final String KEYCLOAK_SETTINGS_ENDPOINT = "http://che-host:8080/api/keycloak/settings";
-    private static final Pattern DISABLED_SETTING_PATTERN = Pattern.compile(".*\"che\\.keycloak\\.disabled\":\"([^ \"]+)\".*");
     
-    private boolean keycloakDisabledRetrieved = false;
-    private boolean keycloakDisabled = false;
+    private Boolean keycloakDisabled = null;
```

Comment:
Well, keycloak settings are expected to be the same for all requests, so in the worst concurrency case, the settings retrieval would be done several times, but with the same value assigned to the `keycloakDisabled` variable. I don't see a real problem here TBH.